### DISCUSSION TITLE
Using ESP-DL

### DISCUSSION THREAD

**Discussion by Farzinkh** [jomjol/AI-on-the-edge-device#2565] [Ideas]
Hi everyone,

Do you have any plan to use [ESP-DL](https://github.com/espressif/esp-dl/tree/fe43a011b312ca7791e1011c32799b3b34223811) to compile the model for espressif SoCs?

**Comment by caco3** [comment]
No, there are no plan.

Be aware that we do no machine learning directly on the device. We do the training offline and then provide the generated tensorflow models which then get run on the device.

**Reply by Farzinkh** [reply]
Exactly ESP-DL has the same strategy the only exception is using it instead of TensorFlow lite. Developers of ESP-Dl claim that it will boost Espressif SoCs performance. However, I am not so sure about that but we can test it. Please check out my repository [AI_EDGE](https://github.com/Farzinkh/AI_EDGE) for better insight into it.

**Reply by caco3** [reply]
I fear migrating to this new library would be more work than our small team currently is able to do.
If you want to look into it, that would be ok and we would try to support you, but I don't think, somebody of the devs has time to look into it.

**Reply by Farzinkh** [reply]
It would be enough if you could provide me the necessary data and model, so I can try to deploy it using this library.
